processor/src/main/java/com/linecorp/decaton/processor/runtime/ProcessorSubscription.java

```diff
@@ -231,14 +234,15 @@ public void run() {
                 commitManager.maybeCommitAsync();
                 metrics.commitOffsetTime.record(timer.duration());
             }
+            updateState(SubscriptionStateListener.State.SHUTTING_DOWN);
+            waitForRemainingTasksCompletion(
+                    scope.props().get(ProcessorProperties.CONFIG_SHUTDOWN_TIMEOUT_MS).value());
```

Comment:
Shouldn't we call `waitForRemainingTasksCompletion` only when `CONFIG_SHUTDOWN_TIMEOUT_MS` is greater than 0?

Since the default value is 0, `waitForRemainingTasksCompletion` could always spit `Timed out` warn logs upon shutdown when users upgraded Decaton version without setting shutdown.timeout.ms (i.e. they don't care re-process at shutdown so warn logs are pointless)

Reply:
Hmm yeah fair enough. Will change.